compiler/rustc_hir_typeck/src/closure.rs

```diff
@@ -832,9 +836,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         bound_sig: ty::PolyFnSig<'tcx>,
     ) -> ClosureSignatures<'tcx> {
         let liberated_sig = self.tcx().liberate_late_bound_regions(expr_def_id, bound_sig);
+        let def_id = self.tcx.hir().opt_local_def_id(body.value.hir_id).unwrap_or(CRATE_DEF_ID);
```

Comment:
```suggestion
        let def_id = self.tcx.hir().body_owner_def_id(body.id());
```
Likewise, this value should be equal to `expr_def_id.expect_owner()`.